aten/src/ATen/core/Device.cpp

```diff
@@ -18,7 +18,28 @@ std::pair<Device::Type, size_t> parse_type(const std::string& device_string) {
   if (position != std::string::npos) {
     return {Device::Type::CUDA, 4};
   }
-  AT_ERROR("Expected 'cpu' or 'cuda' device type at start of device string");
+  position = device_string.find("mkldnn");
```

Comment:
I'm not a fan of this. This was fine when it was two strings, but now this whole function could be reduced by probably 75% if we use a map. Having to count the the length of the string we're looking for is also super fragile. Again, fine when two strings, but now this function warrants a rewrite

Reply:
I think this code still has an opportunity to be faster than an `std` map

Reply:
It could be done with `std::array`, then the overhead should be barely noticeable. I just think everything that changes from `if` clause to `if` clause should be aggregated in some container, and then the logic can be separated from the config, in a sense

Reply:
I'm still slowly remembering C++, but agree this isn't ideal. I'm submitting an update using boost, let me know what you think.

Reply:
Oh we don't have boost in OSS unfortunately

Reply:
ah... ok. already submitted but will discuss offline.